### PR TITLE
add snapshot of node to footer

### PR DESCRIPTION
**Pull request by PfanP**
### Overview

clarity version / commit HEAD initials in webpage footer

### Which JIRA ticket does this PR relate to?

https://casperlabs.atlassian.net/browse/ECO-434

### Complete this checklist before you submit this PR

- [x ] This PR contains no more than 200 lines of code, excluding test code.
- [x ] This PR meets [CasperLabs coding standards](https://casperlabs.atlassian.net/wiki/spaces/EN/pages/16842753/Coding+Standards).
- [ ] If this PR adds a new feature, it includes tests related to this feature.
- [ x] You assigned one person to review this PR.
- [ x] Your GitHub account is linked with our [Drone CI](https://drone-auto.casperlabs.io/) system. This is necessary to run tests on this PR.
- [ x] Do not forget to run `bors r+` if GitHub policy is not enforced, e.g. when merging into another feature branch. It may be omitted under some circumstances if this PR intentionally assumes that integration tests will fail but will be fixed with the future PRs.

### Notes

_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._

### CODE REVIEW

File: packages/ui/src/components/App.tsx [hunk 1]
```diff
@@ -477,27 +477,42 @@ const Alerts = observer((props: AppProps) => {
...
-              rel="noopener noreferrer"
-            >
-              Telegram
-            </a>
-          </small>
+const Footer = () => {
+  const [latestSHA, setLatestSHA] = React.useState(null);
+  React.useEffect(() => {
+    async function myAsyncFunction() {
+      console.log('footer cmd');
```
Comment (AbnerZheng) [inline packages/ui/src/components/App.tsx]: Unused log.

File: packages/ui/src/components/App.tsx [hunk 2]
```diff
@@ -477,27 +477,42 @@ const Alerts = observer((props: AppProps) => {
...
+      const result = await fetch(
+        'https://api.github.com/repos/casperlabs/casper-node/commits/master'
+      );
+      const resultJson = await result.json();
```
Comment (AbnerZheng) [inline packages/ui/src/components/App.tsx]: I don't think this is what we want. You should show the exact version for the currently deployed instance. You could use a library to read from git, and set the environment variable, or there maybe a library that could inject the version before build.

**Review by AbnerZheng — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/App.tsx (hunk 2 above)
Comment (PfanP) [reply packages/ui/src/components/App.tsx]: How could I get the currently deployed version?
The reason I did is once branches are merged to master, drone automatically deploy the result
So I thought the latest version on master is currently deployed

Comment (arcolife) [reply packages/ui/src/components/App.tsx]: The version during build process is to be picked up from the local git branch. The  commit hash of your PR itself in this case would be the version if I would have checked out and built your PR locally. The built server assets would then include the injected version.

Once merged to master, it'll be present in the git log (and if built again later, relying on the deployment pipeline's push/pull based strategy) would include the HEAD. 

This process doesn't need to interact with CI/CD or GitHub APIs. Just local branch HEAD commit hash.

**Comment by zie1ony** [comment]
Hi Khanh. I agree with Abner, that we should have commit that comes from the actual version rather then always `master`.
1. Please review the module suggested by Abner https://www.npmjs.com/package/git-revision-webpack-plugin 
2. Please change the target branch from `master` to `node-rs`.

**Comment by PfanP** [comment]
The only issue while using https://www.npmjs.com/package/git-revision-webpack-plugin package is the clarity UI based on CRA and to config webpack, we should eject it using react-scripts.
So I checked the git-revision-webpack-plugin codebase and used the same git command to get commitHash

**Comment by PfanP** [comment]
bors +

**Pull request closed without merging**